### PR TITLE
Python: [WIP] [local models] Add Support for running local models using Ollama

### PR DESCRIPTION
**Pull request by juharris**
### Motivation and Context
Add Python adapter for getting text completions using models running locally using [Ollama](https://ollama.ai/).

People can now run quick experiments without calling external services by leveraging the convenient tools provided by Ollama to run many different models in a standard way locally.
See the modified notebook for an example. I'll clear outputs before publishing this PR and publish the notebook with outputs somewhere else.

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and **TODO** I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: python/tests/unit/ai/ollama/test_ollama_text_completion.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+import pytest
+
+from semantic_kernel.connectors.ai.complete_request_settings import (
+    CompleteRequestSettings,
+)
+from semantic_kernel.connectors.ai.ollama import (
+    OllamaTextCompletion,
+)
+
+# TODO Update this file for Ollama.
```
Comment (juharris) [inline python/tests/unit/ai/ollama/test_ollama_text_completion.py:15]: Gotta update this file.

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_text_completion.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            )
+
+        if request_settings.logprobs != 0:
+            raise AIException(
+                AIException.ErrorCodes.InvalidRequest,
+                "complete_async does not support logprobs, "
+                f"but logprobs={request_settings.logprobs} was requested",
+            )
+
+        # TODO Set up other custom parameters defined at https://github.com/jmorganca/ollama/blob/main/docs/api.md.
```
Comment (juharris) [inline python/semantic_kernel/connectors/ai/ollama/services/ollama_text_completion.py:140]: Will do this.

**Comment by eavanvalkenburg** [comment]
Overlaps with #4477, @juharris would love your feedback on that one so that we can get ollama in there.

**Comment by juharris** [comment]
#4477 is good enough. Closing

**Pull request closed without merging**